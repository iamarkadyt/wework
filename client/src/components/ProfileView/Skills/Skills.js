import React from 'react'

const skills = ({
  skills
}) => skills && skills.length !== 0 ? (
  <section className="skills">
    <p className="header">Skill Set:</p>
    <div className="skills-list">
      {skills.map((skill, idx) => {
        return (
            <span className="tag"
                key={`skill-${idx}`}>
                {skill}
            </span>
        )
      })}
    </div>
  </section>
) : null

export default skills
