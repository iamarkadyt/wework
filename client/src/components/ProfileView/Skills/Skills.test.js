import React from 'react'
import Skills from './Skills'
import { shallow } from 'enzyme'
import cloneDeep from 'lodash.clonedeep'

describe('Skills', () => {
  let mountedComponent, props

  const getMockProps = () => {
    return cloneDeep({
      skills: [
        "Skill1",
        "Skill2"
      ]
    })
  }

  const comp = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<Skills {...props} />)
    } 
    return mountedComponent
  }

  beforeEach(() => {
    mountedComponent = undefined
    props = getMockProps()
  })

  it('matches snapshot', () => {
    expect(comp()).toMatchSnapshot()
  })
})
