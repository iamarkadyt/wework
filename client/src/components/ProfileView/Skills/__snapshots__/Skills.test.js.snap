// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills matches snapshot 1`] = `
<section
  className="skills"
>
  <p
    className="header"
  >
    Skill Set:
  </p>
  <div
    className="skills-list"
  >
    <span
      className="tag"
      key="skill-0"
    >
      Skill1
    </span>
    <span
      className="tag"
      key="skill-1"
    >
      Skill2
    </span>
  </div>
</section>
`;
